core/src/main/scala/zio/jdbc/SqlFragment.scala

```diff
@@ -114,50 +102,116 @@ final class Sql[+A](
     s"Sql(${sql.result()}$paramsString)"
   }
 
-  def values[B](bs: Iterable[B])(implicit encode: JdbcEncoder[B], ev: IsSqlFragment[A]): SqlFragment =
+  def values[B](bs: Iterable[B])(implicit encoder: JdbcEncoder[B]): SqlFragment =
     this ++
-      Sql.values ++
-      Sql.intersperse(
-        Sql.comma,
-        bs.map(b => Sql.lparen ++ encode.encode(b) ++ Sql.rparen)
+      SqlFragment.values ++
+      SqlFragment.intersperse(
+        SqlFragment.comma,
+        bs.map(b => SqlFragment.lparen ++ encoder.encode(b) ++ SqlFragment.rparen)
       )
 
   def valuesBatched[B](bs: Iterable[B], batchSize: Int = 2000, tableName: String)(
     keys: String*
-  )(implicit encode: JdbcEncoder[B], ev: IsSqlFragment[A]): Seq[SqlFragment] = {
+  )(implicit encoder: JdbcEncoder[B]): Seq[SqlFragment] = {
     val batches          = bs.grouped(batchSize)
-    val insertStatements = batches.map(batch => Sql.insertInto(tableName)(keys.mkString(", ")).values(batch)).toSeq
+    val insertStatements =
+      batches.map(batch => SqlFragment.insertInto(tableName)(keys.mkString(", ")).values(batch)).toSeq
     insertStatements
   }
 
-  def values[B](b: B, bs: B*)(implicit encoder: JdbcEncoder[B], ev: IsSqlFragment[A]): SqlFragment = values(b +: bs)
+  def values[B](b: B, bs: B*)(implicit encoder: JdbcEncoder[B]): SqlFragment =
+    values(b +: bs)
+
+  def where(predicate: SqlFragment): SqlFragment =
+    self ++ SqlFragment.where ++ predicate
+
+  def query[A](implicit decoder: JdbcDecoder[A]): Query[A] =
```

Comment:
Generally, to aid code organization and discoverability, all constructors for `X` should be located in the companion object for `X`. Can you move this and others as appropriate to follow this convention?